src/new/sendTransaction.ts

```diff
@@ -422,6 +424,23 @@ export default class SendTransaction extends EventEmitter {
               throw new WalletError(ErrorMessages.TRANSACTION_IS_NULL);
             }
             this.transaction.updateHash();
+            if (this.storage) {
```

Comment:
This is the core of the PR.

When we push a transaction we need to convert it from a `Transaction` instance to a `IHistoryTx` (which is what is received via websocket) and add it to the storage with normal processing.

If we are creating a token we should also add it to the tokens here because the fullnode may not know that this token exists when we ask for the name and symbol for this UID. It also saves us from making an extraneous API call.